Dockerfile

```diff
@@ -0,0 +1,73 @@
+FROM golang:1.10-alpine AS build
+
+RUN apk add --update \
+      curl \
+      git \
+      make \
+      py-pip \
+      python \
+      python-dev \
+      && true
+
+RUN mkdir /out
+RUN mkdir -p /out/etc/apk && cp -r /etc/apk/* /out/etc/apk/
+RUN apk add --no-cache --initdb --root /out \
+    alpine-baselayout \
+    busybox \
+    ca-certificates \
+    coreutils \
+    git \
+    libc6-compat \
+    libgcc \
+    libstdc++ \
+    python \
+    && true
+
+ENV SKAFFOLD $GOPATH/src/github.com/GoogleContainerTools/skaffold
+RUN mkdir -p "$(dirname ${SKAFFOLD})"
+COPY . $SKAFFOLD
+
+WORKDIR $SKAFFOLD
+RUN make \
+    && cp out/skaffold /out/usr/local/bin/skaffold
+
+WORKDIR /out
+
+RUN ln -s /lib /lib64
+
+ENV KUBECTL_VERSION v1.10.6
+RUN curl --silent --location "https://dl.k8s.io/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" --output usr/local/bin/kubectl \
+    && chmod +x usr/local/bin/kubectl
+
+ENV DOCKER_VERSION 18.03.0
```

Comment:
One constraint I have here, is that it has to support multi-stage build, and I'm happy to include the latest version available (perhaps LTS one), but I don't know if we have other constrains or not...